documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -573,6 +577,43 @@ junit.jupiter.testmethod.order.default = \
 Similarly, you can specify the fully qualified name of any custom class that implements
 `MethodOrderer`.
 
+==== Class Order
+
+For build optimization purposes, it can be helpful to run _test classes_ in a specific
+order.
+For example:
+* previously failed tests and faster tests first: fail fast
+* with parallelism enabled, longer tests first: shortest test execution duration
+* etc.
+
+You can use the `junit.jupiter.testclass.order.default`
+<<running-tests-config-params, configuration parameter>> to specify the fully qualified
+class name of the `{ClassOrderer}` you would like to use. The supplied class has to
+implement the `ClassOrderer` interface.
+
+You can implement your own custom `ClassOrderer` or use one of the
+following built-in `ClassOrderer` implementations.
+
+* `{DisplayName}`: sorts test classes _alphanumerically_ based on their display names
+  (see <<writing-tests-display-name-generator-precedence-rules, display name generation
+  precedence rules>>).
+* `{ClassName}`: sorts test classes _alphanumerically_ based on their class name.
+* `{OrderAnnotation}`: sorts test classes _numerically_ based on values specified via the
+  `{Order}` annotation.
+* `{Random}`: orders test classes _pseudo-randomly_ and supports configuration of a
+  custom _seed_.
+
+For example for the `@Order` annotation to be honored on _test classes_, you should set
+the `{OrderAnnotation}` class orderer, using the configuration parameter with the
+corresponding fully qualified class name (e.g., in
+`src/test/resources/junit-platform.properties`):
+
+[source,properties,indent=0]
+----
+junit.jupiter.testclass.order.default = \
+    org.junit.jupiter.api.ClassOrderer$OrderAnnotation
```

Comment:
I'm wondering whether we should allow setting `junit.jupiter.testclass.order.default = OrderAnnotation` as a shorthand and do the same for `junit.jupiter.testmethod.order.default` and `junit.jupiter.displayname.generator.default`. For backward compatibility, we could first attempt to load the class and, if that fails, try again with the prepended name, e.g. `org.junit.jupiter.api.ClassOrderer$`. Thought?

Reply:
I think that most people going that deep into customization will have no trouble using a fully qualified class name.
Besides this behavior can be added in the future if needed.

Reply:
Agreed.